llms/mlx_lm/server.py

```diff
@@ -329,26 +334,36 @@ def handle_stream(
                     tokens = tokens[: -stop_condition.trim_length]
                 break
 
+            # "\ufffd" is used to indicate to the tokenizer, that subsequent characters
+            # should be combined into a single unicode character
+            if "\ufffd" in TOKENIZER.decode(token):
+                continue
+
+            # If the end of tokens overlaps with a stop sequence
+            # Generate new tokens until we know if the stop sequence is hit or not
+            if any(
+                (sequence_overlap(tokens, sequence) for sequence in stop_id_sequences)
+            ):
+                continue
+
             # Workaround for llama tokenizer emitting spaces when decoding token by token.
             generated_text = TOKENIZER.decode(tokens)
-            new_text = generated_text[current_generated_text_index:]
-            current_generated_text_index = len(generated_text)
+            new_text = generated_text[current_text_index:]
+            current_text_index = len(generated_text)
 
             response = self.generate_response(new_text, None)
             self.wfile.write(f"data: {json.dumps(response)}\n\n".encode())
             self.wfile.flush()
-            stop_sequence_buffer = []
-
-        # check is there any remaining text to send
-        if stop_sequence_buffer:
-            generated_text = TOKENIZER.decode(tokens)
-            next_chunk = generated_text[current_generated_text_index:]
-            response = self.generate_response(next_chunk, "length")
 
+        # Check for remaining text
+        generated_text = TOKENIZER.decode(tokens)
+        if len(generated_text) > current_text_index:
+            new_text = generated_text[current_text_index:]
+            response = self.generate_response(new_text, "length")
             self.wfile.write(f"data: {json.dumps(response)}\n\n".encode())
             self.wfile.flush()
 
-        self.wfile.write("data: [DONE]\n\n".encode())
+        self.wfile.write(f"data: [DONE]\n\n".encode())
```

Comment:
why make that a format string?

Reply:
Ah, I think I pasted it from the old server.py before my last PR got merged that removed it. I'll remove it